### ISSUE TITLE
orderByDistance  -return the sorted array

### ISSUE THREAD

**Issue by alastair-todd** [manuelbieh/Geolib#74]
Why does the function orderByDistance return a new array with a different shape to the original?

Convention says this method should return an array with the same contents as the source but sorted by distance.

**Comment by manuelbieh** [comment]
Good point!

**Comment by alastair-todd** [comment]
Did my own :)

 ```
/**
    * Sorts an array of coords by distance from a reference coordinate
    *
    * @param        object      reference coordinate e.g. {latitude: 51.5023, longitude: 7.3815}
    * @param        mixed       array or object with coords [{latitude: 51.5143, longitude: 7.4138}, {latitude: 123, longitude: 123}, ...]
    * @return       array       a copy of the original array ordered by distance
    */
    var sortByDistance = function(latlng, coords) {

        var coordsArray = [];

        for(var index = 0, len = coords.length; index < len; index++) {
            var d = geolib.getDistance(latlng, coords[index]);
            coords[index].distance = d;
            coordsArray.push(coords[index]);

        }
        return coordsArray.sort(function(a, b) { return a.distance - b.distance; });

    }
```

**Comment by manuelbieh** [comment]
This should work, yes :)

Plus:
`coords[index].key = coord;`

Can you add it and make a pull request? Otherwise I will do i later

**Comment by manuelbieh** [comment]
This is fixed in v2.0.19!
